Sources/DistributedSystem/DistributedSystem.swift

```diff
@@ -25,6 +25,41 @@ extension Channel {
     }
 }
 
+public enum CompressionMode {
+    public struct Dictionary {
+        let data: UnsafeRawBufferPointer
+        let checksum: UInt32
+
+        // very simple checksum calculation
```

Comment:
Maybe use e.g. CRC32, according to ChatGPT it has rather simple implementation:
```swift
import Foundation

/// Computes the CRC32 checksum for the given buffer.
/// - Parameter buffer: The data buffer for which the CRC32 checksum is computed.
/// - Returns: The CRC32 checksum as a `UInt32`.
func crc32(buffer: Data) -> UInt32 {
    let polynomial: UInt32 = 0xEDB88320
    var crc: UInt32 = 0xFFFFFFFF

    for byte in buffer {
        var currentByte = UInt32(byte)
        for _ in 0..<8 {
            let mix = (crc ^ currentByte) & 1
            crc >>= 1
            if mix != 0 {
                crc ^= polynomial
            }
            currentByte >>= 1
        }
    }

    return ~crc
}

// Example usage
let data = "Hello, World!".data(using: .utf8)!
let checksum = crc32(buffer: data)
print(String(format: "CRC32: %08X", checksum))
```

Verified it for some examples using https://crc32.online/ :)

Reply:
Your code works with `Data` type, should be adopted to UnsafeRawBufferPointer.
Let's leave as is for now :)

Reply:
```swift
import Foundation

/// Computes the CRC32 checksum for the given buffer.
/// - Parameter buffer: The raw buffer for which the CRC32 checksum is computed.
/// - Returns: The CRC32 checksum as a `UInt32`.
func crc32(buffer: UnsafeRawBufferPointer) -> UInt32 {
    let polynomial: UInt32 = 0xEDB88320
    var crc: UInt32 = 0xFFFFFFFF

    for byte in buffer {
        var currentByte = UInt32(byte)
        for _ in 0..<8 {
            let mix = (crc ^ currentByte) & 1
            crc >>= 1
            if mix != 0 {
                crc ^= polynomial
            }
            currentByte >>= 1
        }
    }

    return ~crc
}

// Example usage
let string = "Hello, World!"
string.withCString { cString in
    let length = strlen(cString)
    let buffer = UnsafeRawBufferPointer(start: cString, count: length)
    let checksum = crc32(buffer: buffer)
    print(String(format: "CRC32: %08X", checksum)) // Output: CRC32: 1C291CA3
}
```

Enjoy!